### PR TITLE
feat: support junit output for unit/e2e tests

### PR DESCRIPTION
**Pull request by jaredallard**
**What this PR does / why we need it**: Title says, it does. 

**JIRA ID**: [DTSS-0]

**Notes for your reviewer**:

### CODE REVIEW

**Review by george-e-shaw-iv — CHANGES_REQUESTED** [review]
adds too much code, try and simplify this please

File: shell/test.sh [hunk 1]
```diff
@@ -128,6 +128,11 @@ set -ex
 # Why: We want these to split. For those wondering about "$@":
 # https://stackoverflow.com/questions/5720194/how-do-i-pass-on-script-arguments-that-contain-quotes-spaces
 # shellcheck disable=SC2086
-go test $BENCH_FLAGS $COVER_FLAGS $TEST_FLAGS \
+"$DIR/gobin.sh" gotest.tools/gotestsum@v1.7.0 --junitfile bin/unit-tests.xml -- $BENCH_FLAGS $COVER_FLAGS $TEST_FLAGS \
   -ldflags "-X github.com/getoutreach/go-outreach/v2/pkg/app.Version=testing -X github.com/getoutreach/gobox/pkg/app.Version=testing" -tags="$TEST_TAGS" \
   "$@" ./...
+
+if [[ -n $CI ]]; then
```
Comment (masterkidan) [inline shell/test.sh:135]: just for understanding: can you add a comment here as to why this move in file directory is required? Guessing it has to do with circle ci , but would be good to explicitly call it out.

File: shell/test.sh [hunk 2]
```diff
@@ -128,6 +128,11 @@ set -ex
 # Why: We want these to split. For those wondering about "$@":
 # https://stackoverflow.com/questions/5720194/how-do-i-pass-on-script-arguments-that-contain-quotes-spaces
 # shellcheck disable=SC2086
-go test $BENCH_FLAGS $COVER_FLAGS $TEST_FLAGS \
+"$DIR/gobin.sh" gotest.tools/gotestsum@v1.7.0 --junitfile bin/unit-tests.xml -- $BENCH_FLAGS $COVER_FLAGS $TEST_FLAGS \
```
Comment (masterkidan) [inline shell/test.sh:131]: Rather than lock it to 1.7.0 explicitly mind putting this in an environment variable? Just makes it more configurable

**Review by masterkidan — APPROVED** [review]

File: shell/test.sh (hunk 2 above)
Comment (vk-outreach) [reply shell/test.sh:131]: We don't really use environment variables (and we don't want to start doing that either).  

I think we should use the service.yaml versions override setup for this like we do for other tools.

Comment (jaredallard) [reply shell/test.sh:131]: Fixed

File: shell/test.sh (hunk 1 above)
Comment (jaredallard) [reply shell/test.sh:136]: Fixed

**Review by george-e-shaw-iv — APPROVED** [review]

**Pull request merged**